server/oceandbs/views.py

```diff
@@ -597,3 +596,100 @@ def get(self, request):
                 "type": quote.storage.type,
                 "CID": json.loads(response.content)[0]['CID']
             })
+
+class QuoteHistory(APIView):
+    @csrf_exempt
+    @extend_schema(
+        request=[],
+        parameters=[
+            OpenApiParameter(
+                name='userAddress',
+                description='User Address',
+                type=str
+            ),
+            OpenApiParameter(
+                name='nonce',
+                description='Nonce',
+                type=int
+            ),
+            OpenApiParameter(
+                name='signature',
+                description='Signature',
+                type=str
+            )
+        ],
+        examples=[
+            OpenApiExample(
+                "QuoteHistoryResponseExample",
+                value=[
+                    {
+                        "quoteId": "23",
+                        "status": 400,
+                        "chainId": 80001,
+                        "tokenAddress": "0x222",
+                        "tokenAmount": "999999999",
+                        "approveAddress": "0x1234",
+                        "transactionHash": "xxxx"
+                    },
+                    {
+                        "quoteId": "23",
+                        "chainId": 80001,
+                        "tokenAddress": "0x222",
+                        "tokenAmount": "999999999",
+                        "approveAddress": "0x1234",
+                        "requestId": "xxxx"
+                    }
+                ],
+                request_only=False,
+                response_only=True
+            )
+        ],
+        responses={
+            200: inline_serializer(
+                name='QuoteHistoryResponseSerializer',
+                fields={
+                    "quoteId": serializers.CharField(),
+                    "status": serializers.IntegerField(),
+                    "chainId": serializers.IntegerField(),
+                    "tokenAddress": serializers.CharField(),
+                    "tokenAmount": serializers.CharField(),
+                    "approveAddress": serializers.CharField(),
+                    "transactionHash": serializers.CharField(),
+                    "requestId": serializers.CharField(),
+                }
+            ),
+            404: OpenApiResponse(description='History Not Found.'),
+        }
+    )
+    def get(self, request):
+        params = {**request.GET}
+
+        if not all(key in params for key in ('userAddress', 'nonce', 'signature')):
```

Comment:
We could improve this by also checking that each parameter is of the right type. It would also be helpful to improve the response to mention which parameter is missing in the response